integration/kubernetes/helm-chart/alluxio/templates/config/alluxio-conf.yaml

```diff
@@ -32,15 +32,141 @@ metadata:
     heritage: {{ .Release.Service }}
 data:
   ALLUXIO_JAVA_OPTS: |-
-    {{if $isSingleMaster}}-Dalluxio.master.hostname={{ $fullName }}-{{ $defaultMasterName }}{{end}} -Dalluxio.master.journal.type={{ .Values.journal.type }} -Dalluxio.master.journal.folder={{ .Values.journal.folder }} {{if $isHaEmbedded}}-Dalluxio.master.embedded.journal.addresses={{range $i := until $masterCount}}{{ $fullName }}-master-{{$i}}:19200,{{end}}{{end}} {{ .Values.jvmOptions }} {{- range $key, $val := .Values.properties }} -D{{ $key }}={{ $val }} {{ end }}
+    {{- $alluxioJavaOpts := list }}
+
+    {{- /* Specify master hostname if single master */}}
+    {{- if $isSingleMaster }}
+      {{- $alluxioJavaOpts = printf "-Dalluxio.master.hostname=%v-%v" $fullName $defaultMasterName | append $alluxioJavaOpts }}
```

Comment:
this seems like a redundant pattern, can we have a builder for java opts?

Reply:
Sadly Helm doesn't have good support of customized functions other than what's provided in http://masterminds.github.io/sprig/.
This can't be easily done :(